README.md
# stay-awake

<p align="center">
  <a href="https://pypi.org/project/stay-awake/">
    <img src="https://raw.githubusercontent.com/singhsidhukuldeep/stay-awake/main/images/sample.PNG" width="1080" alt="Go to https://pypi.org/project/stay-awake/">
  </a>
</p>
<p align="center">
<a href="https://pypi.org/project/stay-awake/"><img src="https://img.shields.io/pypi/pyversions/stay-awake" alt="Go to https://pypi.org/project/stay-awake/"/></a>
<a href="https://pypi.org/project/stay-awake/"><img src="https://img.shields.io/pypi/v/stay-awake" alt="Go to https://pypi.org/project/stay-awake/"/></a>
<a href="https://pypi.org/project/stay-awake/"><img src="https://img.shields.io/pypi/status/stay-awake" alt="Go to https://pypi.org/project/stay-awake/"/></a>
<a href="https://pypi.org/project/stay-awake/"><img src="https://img.shields.io/pypi/format/stay-awake" alt="Go to https://pypi.org/project/stay-awake/"/></a>
<a href="https://pypistats.org/packages/stay-awake"><img src="https://img.shields.io/pypi/dm/stay-awake"/></a>
</p>

Stay-Awake is a Simple Platform Independent Python package to keep your system awake without affecting workflow!

* Does this affect workflow? 
> No, this only get's triggered when you don't do any mouse movements!

* Is there a GUI?
> This was intended to be a light weight solution, so as of now it only has a CLI!

* How does it work?
> If in a span of 60 seconds you don't move your mouse, this script will automatically move your mouse for about 1 to 4 pixels randomly. There won't be any mouse displacement! If you are working, this will do absolutely nothing!

## Current Status

[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://github.com/singhsidhukuldeep)
[![GitHub up-time BOT](https://badgen.net/uptime-robot/month/ur967659-422c6e77bfb79bb6a47c642c)](https://github.com/singhsidhukuldeep/stay-awake/deployments/activity_log?environment=github-pages)
[![Commits](https://img.shields.io/github/last-commit/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/commits/master)
[![DependaBot](https://api.dependabot.com/badges/status?host=github&repo=singhsidhukuldeep/stay-awake)](https://dependabot.com/)

[![Issues](https://img.shields.io/github/issues/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/issues)
[![Total Commits](https://badgen.net/github/commits/singhsidhukuldeep/stay-awake/main)](https://github.com/singhsidhukuldeep/stay-awake/commits/main)
[![Contributors](https://badgen.net/github/contributors/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/graphs/contributors)
[![Forks](https://badgen.net/github/forks/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/network/members)
[![Stars](https://badgen.net/github/stars/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/stargazers)
[![Watchers](https://badgen.net/github/watchers/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/watchers)
[![Branches](https://badgen.net/github/branches/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake/branches)

[![License: AGPL v3](https://img.shields.io/badge/License-AGPL%20v3-blue.svg)](https://www.gnu.org/licenses/agpl-3.0)
[![made-with-python](https://img.shields.io/badge/Made%20with-Python3.5+-1f425f.svg)](https://www.python.org/)
[![made-with-Markdown](https://img.shields.io/badge/Made%20with-Markdown-1f425f.svg)](http://commonmark.org)
[![repo- size](https://img.shields.io/github/repo-size/singhsidhukuldeep/stay-awake)](https://github.com/singhsidhukuldeep/stay-awake)
[![Followers](https://img.shields.io/github/followers/singhsidhukuldeep?style=plastic&logo=github)](https://github.com/singhsidhukuldeep?tab=followers)

## Setup

### Installing Package
 
 ```shell
pip3 install stay-awake
 ```

### Running

```shell
python3 -m stay-awake
```

> You can also give custom timeouts
> Eg: for 5 minutes (default is 1 min) `python3 -m stay-awake 5`

### Virtual Environment

**Important:** *Virtual Environment is recommended*

* **`Linux`** Systems:

    ```shell
    python3 -m venv ./venv

    source venv/bin/activate

    pip3 install -r requirements.txt
    ```

    ```shell
    deactivate
    ```

* **`Windows`** Systems:

    ```shell
    python3 -m venv ./venv

    venv\Scripts\activate

    pip3 install -r requirements.txt
    ```

    ```shell
    venv\Scripts\deactivate
    ```

    If getting issue in installing virtualenv on `windows`, use administrator privileges
    


## Steps for publishing to `pypi` [This is just for me, Maybe!]

- `pip3 install setuptools twine`
- Go to project folder
- `python3 setup.py sdist`
- `twine upload --repository-url https://upload.pypi.org/legacy/ dist/*`

OR

Go to your project folder and:
```shell
pip3 install setuptools twine

python3 setup.py sdist
twine upload --repository-url https://upload.pypi.org/legacy/ dist/*
```

## Credits

### Maintained by

***Kuldeep Singh Sidhu*** 

Github: [github/singhsidhukuldeep](https://github.com/singhsidhukuldeep)
`https://github.com/singhsidhukuldeep`

Website: [Kuldeep Singh Sidhu (Website)](http://kuldeepsinghsidhu.com)
`http://kuldeepsinghsidhu.com`

LinkedIn: [Kuldeep Singh Sidhu (LinkedIn)](https://www.linkedin.com/in/singhsidhukuldeep/)
`https://www.linkedin.com/in/singhsidhukuldeep/`

### Contributors

The full list of all the contributors is available [here](https://github.com/singhsidhukuldeep/stay-awake/graphs/contributors)


[![website](https://forthebadge.com/images/badges/built-with-love.svg)](http://kuldeepsinghsidhu.com)

## Say Thanks

If this helped you in any way, it would be great if you could share it with others.


CHANGELOG.md
CHANGE LOG
==============================

0.1 (24/10/2020)
------------------------
- Released first version(stable) of stay-awake.

0.2 (27/11/2020)
------------------------
- Second stable version
- Added absolute control by allowing to interrupt the process mid-way.
- Added functionality to add custom wait time.

0.2.1 (27/11/2020)
------------------------
- Fixed Issues

0.3 (27/11/2020)
------------------------
- Fixed Issues in `.md` files

stay-awake/__init__.py
# !/usr/bin/env python3
# coding:utf-8
"""
Name       : __init__.py
Author     : Kuldeep Singh Sidhu
Time       : 27-11-2020 02:58 PM
GitHub     : https://github.com/singhsidhukuldeep
Description: The main module that is used for triggering stay-awake
"""

print('>> Starting')

userName = "singhsidhukuldeep"
repoName = "stay-awake"
linkedIn = "https://www.linkedin.com/in/singhsidhukuldeep/"
website = "http://kuldeepsinghsidhu.com"
githubLink = f"https://github.com/{userName}"
name = "Kuldeep Singh Sidhu"
license = "agpl-3.0"
repoLink = f"{githubLink}/{repoName}"
errorMessage = f">> Report any issues at {repoLink}"

print(f"\n\n>> *-* Thank You for using Stay-Awake, please STAR this repo at {repoLink} ! *-* <<\n\n")

# print(errorMessage)

print('>> Importing Libraries')
try:
    import pyautogui
    import time
    import sys
    from datetime import datetime
    from random import randint
    from tqdm import tqdm
except Exception as exp:
    print (f'>><< Error: {exp}')
    print(errorMessage)
    exit()
print('>> Libraries Imported')

print('>> Setting Up')
try:
    pyautogui.FAILSAFE = False
    waitTime = 1
    try:
        waitTime = max(1, int(sys.argv[-1]))
    except Exception as exp:
        pass
    prevousLocation = None
    print(f'>> Wait time set to: {waitTime} minutes')
except Exception as exp:
    print (f'>><< Error: {exp}')
    print(errorMessage)
    exit()
print(f'>> Setup Complete')

def hasMoved(currentLocation):
    try:
        time.sleep(randint(0,2))
        if pyautogui.position() == currentLocation:
            return False
        else:
            print('>> Input detected, interrupting stay-awake')
            return True
    except Exception as exp:
        print(f'>><< Error: {exp}')
        print(errorMessage)
        exit()

def doMove(currentLocation):
    try:
        print(f'>> Moving at {currentLocation}')
        for n_move in range(1, randint(2,4)):
            if hasMoved(currentLocation):break
            pyautogui.moveTo(currentLocation[0] + n_move, currentLocation[1] + n_move)
            pyautogui.moveTo(currentLocation)
            if hasMoved(currentLocation): break
            pyautogui.moveTo(currentLocation[0] - n_move, currentLocation[1] - n_move)
            pyautogui.moveTo(currentLocation)
            if hasMoved(currentLocation): break
            pyautogui.moveTo(currentLocation[0] - n_move, currentLocation[1] + n_move)
            pyautogui.moveTo(currentLocation)
            if hasMoved(currentLocation): break
            pyautogui.moveTo(currentLocation[0] + n_move, currentLocation[1] - n_move)
            pyautogui.moveTo(currentLocation)
        print(f'>> Made movement at {datetime.now().time()}')
    except Exception as exp:
        print(f'>><< Error: {exp}')
        print(errorMessage)
        exit()

try:
    while (True):
        print('>> Checking')
        if prevousLocation == None:
            prevousLocation = pyautogui.position()
            print('>> Testing Stay-Awake')
            doMove(prevousLocation)
        elif prevousLocation == pyautogui.position():
            print(">> No manual movement detected >> Triggering stay-awake")
            doMove(prevousLocation)
        else:
            currentLocation = pyautogui.position()
            print(f'>> Mouse manually moved from {prevousLocation} to {currentLocation} >> No stay-awake triggered')
            prevousLocation = currentLocation
        for _ in tqdm(range(waitTime * 60), desc = f'>> Waiting for {waitTime*60} seconds'):
            time.sleep(1)
except Exception as exp:
    print(f'>><< Error: {exp}')
    print(errorMessage)
    exit()